addOns/authhelper/src/test/java/org/zaproxy/addon/authhelper/report/ExtensionAuthhelperReportUnitTest.java

```diff
@@ -0,0 +1,207 @@
+ * limitations under the License.
+ */
+package org.zaproxy.addon.authhelper.report;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.mockito.Mockito.CALLS_REAL_METHODS;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.withSettings;
+
+import java.io.File;
+import java.nio.file.Files;
+import net.sf.json.JSONArray;
+import net.sf.json.JSONObject;
+import org.apache.commons.text.StringEscapeUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.quality.Strictness;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.ExtensionLoader;
+import org.parosproxy.paros.model.Model;
+import org.zaproxy.addon.authhelper.ExtensionAuthhelper;
+import org.zaproxy.addon.reports.ExtensionReports;
+import org.zaproxy.addon.reports.ReportData;
+import org.zaproxy.addon.reports.Template;
+import org.zaproxy.zap.extension.alert.AlertNode;
+import org.zaproxy.zap.testutils.TestUtils;
+import org.zaproxy.zap.utils.ZapXmlConfiguration;
+
+class ExtensionAuthhelperReportUnitTest extends TestUtils {
+
+    @BeforeEach
+    void setUp() throws Exception {
+        mockMessages(new ExtensionAuthhelper());
+
+        Model model = mock(Model.class, withSettings().defaultAnswer(CALLS_REAL_METHODS));
+        Model.setSingletonForTesting(model);
+        ExtensionLoader extensionLoader =
+                mock(ExtensionLoader.class, withSettings().strictness(Strictness.LENIENT));
+        Control.initSingletonForTesting(Model.getSingleton(), extensionLoader);
+        Model.getSingleton().getOptionsParam().load(new ZapXmlConfiguration());
+
+        Constant.PROGRAM_VERSION = "Test Build";
+    }
+
+    private ReportData getGenericReportData(String templateName) {
+        ReportData reportData = new ReportData(templateName);
+        reportData.setTitle("Test Title");
+        reportData.setDescription("Test Description");
+        reportData.setIncludeAllConfidences(true);
+        reportData.setIncludeAllRisks(true);
+        reportData.setAlertTreeRootNode(new AlertNode(0, "Test"));
+        return reportData;
+    }
+
+    @ParameterizedTest
+    @ValueSource(
+            strings = {
+                "https://example.com",
+                "https://example.com",
+                "https://example.com.*",
+                "https://example.com/",
+                "https://example.com/.*",
+                "https://example.com/test123",
+                "https://example.com/test123.*",
+                "https://example.com?param",
+            })
```

Comment:
I gather this is accommodating regex patterns. Add or modify something with:
`"https?://example.com"`
`"https://.*example\.com.*"`

??

Reply:
Yes my descriptions are probably imprecise but hopefully you get the idea ☺️

Reply:
Confession - this is a bit of a hack as we know there are cases it wont handle. But it should handle the common cases. The problem is that the context doesnt record the URLs - someone could def define a set in include regexes that this wont handle